packages/shared/src/utils/validation/declarative-validation-rules.ts

```diff
@@ -56,6 +56,10 @@ const validRegular = (value: string, options: TOptions) => options.regex?.test(v
 const confirmRequired = (value: string) => !!value;
 const checkPOBox = (value: string) => !/p[.\s]+o[.\s]+box/i.test(value);
 const validEmailToken = (value: string) => value.trim().length === 8;
+export const hasInvalidCharacters = (target_string: string) => /[^\dX\s]/.test(target_string);
+export const isFormattedCardNumber = (target_string: string) =>
+    /(^\d{4})\s(\d{2}X{2})\s(X{4})\s(\d{4}$)/.test(target_string);
```

Comment:
Isn't is supposed to come from the API?

Reply:
This is a RegExp for formatted text. May be we can start a discussion regd moving these validations to BE to keep them in sync